kmir/src/kmir/kdist/mir-semantics/kmir.k

```diff
@@ -14,167 +15,717 @@ module KMIR-SYNTAX
+
+  rule #evalArgs(rvalueAddressOf(_MUT, _PLACE))                                      => .K
+  rule #evalArgs(rvalueAggregate(_KIND, ARGS))                                       => #evalArgs(#argsToList(ARGS, .List), .List)
+  rule #evalArgs(rvalueBinaryOp(_OP, ARG1, ARG2))                                    => #evalArgs(ListItem(ARG1) ListItem(ARG2), .List)
+  rule #evalArgs(rvalueCast(_KIND, ARG, _TY))                                        => #evalArgs(ListItem(ARG), .List)
+  rule #evalArgs(rvalueCheckedBinaryOp(_OP, ARG1, ARG2))                             => #evalArgs(ListItem(ARG1) ListItem(ARG2), .List)
+  rule #evalArgs(rvalueCopyForDeref(PLACE))                                          => #evalArgs(ListItem(PLACE), .List)
+  rule #evalArgs(rvalueDiscriminant(PLACE))                                          => #evalArgs(ListItem(PLACE), .List)
+  rule #evalArgs(rvalueLen(PLACE))                                                   => #evalArgs(ListItem(PLACE), .List)
+  rule #evalArgs(rvalueRef(_REG, _BRW_KIND, _PLACE))                                 => .K
+  rule #evalArgs(rvalueRepeat(ARG, CONST))                                           => #evalArgs(ListItem(ARG) ListItem(CONST), .List)
+  rule #evalArgs(rvalueShallowInitBox(ARG, _TY))                                     => #evalArgs(ListItem(ARG), .List)
+  rule #evalArgs(rvalueThreadLocalRef(_CRATE_ITEM))                                  => .K
+  rule #evalArgs(rvalueNullaryOp(_OP, _TY))                                          => .K
+  rule #evalArgs(rvalueUnaryOp(_OP, ARG))                                            => #evalArgs(ListItem(ARG), .List)
+  rule #evalArgs(rvalueUse(ARG))                                                     => #evalArgs(ListItem(ARG), .List)
+  rule #evalArgs(nonDivergingIntrinsicAssume(ARG))                                   => #evalArgs(ListItem(ARG), .List)
+  rule #evalArgs(nonDivergingIntrinsicCopyNonOverlapping(ARGS))                      => #evalArgs(ListItem(src(ARGS)) ListItem(dst(ARGS)) ListItem(count(ARGS)), .List)
+  rule #evalArgs(terminatorKindSwitchInt(ARG, _TARGETS))                             => #evalArgs(ListItem(ARG), .List)
+  rule #evalArgs(terminatorKindResume)                                               => .K
+  rule #evalArgs(terminatorKindAbort)                                                => .K
+  rule #evalArgs(terminatorKindReturn)                                               => .K
+  rule #evalArgs(terminatorKindUnreachable)                                          => .K
+  rule #evalArgs(terminatorKindDrop(_PLACE, _TARGET, _ACT))                          => .K
+  rule #evalArgs(terminatorKindCall(FUNC, ARGS, _DEST, _TARGET, _ACT))               => #evalArgs(#argsToList(FUNC ARGS, .List), .List)
+  rule #evalArgs(assert(COND, _EXPECTED, _MSG, _TARGET, _ACT))                       => #evalArgs(ListItem(COND), .List)
+  rule #evalArgs(terminatorKindInlineAsm(_TMPL, _ARGS, _OPTS, _SPNS, _TARGET, _ACT)) => .K
```

Comment:
I think we need a rule like these `#evalArgs(terminatorKindGoto) -> .K`, otherwise we will get stuck when trying to evaluate the exec rule for the goto terminator.

Reply:
This is a bug. Will fix.